xml/admin_troubleshooting.xml

```diff
@@ -0,0 +1,318 @@
+&prompt.root.master;<command>kubectl cluster-info</command>
+&prompt.root.master;<command>journalctl -u kubelet</command>
+&prompt.root.master;<command>journalctl -u kube-apiserver</command>
+&prompt.root.master;<command>journalctl -u kube-scheduler</command>
+&prompt.root.master;<command>journalctl -u kube-controller-manager</command></screen>
+   </step>
+   <step>
+    <para>
+     Collect all relevant logs with <command>supportconfig</command> on
+     the &master_node; and &admin_node;.
+    </para>
+<screen>&prompt.root.admin;<command>supportconfig</command>
+&prompt.root.admin;<command>tar -xvjf /var/log/nts_*.tbz</command>
+&prompt.root.admin;<command>cd nts*</command>
+&prompt.root.admin;<command>cat etcd.txt kubernetes.txt salt-minion.txt</command></screen>
+<screen>&prompt.root.master;<command>supportconfig</command>
+&prompt.root.master;<command>tar -xvjf /var/log/nts_*.tbz</command>
+&prompt.root.master;<command>cd nts*</command>
+&prompt.root.master;<command>cat etcd.txt kubernetes.txt salt-minion.txt</command></screen>
+   </step>
+  </procedure>
+ </sect1>
+
+ <sect1 xml:id="sec.admin.troubleshooting.failed_update">
+  <title>Recovering from Failed Update</title>
+  <para>
+   See <xref
+   linkend="sec.admin.software.transactional-updates.recovering" />.
+  </para>
+ </sect1>
+
+ <sect1 xml:id="sec.admin.troubleshooting.etcd">
+  <title>Checking <literal>etcd</literal> Health</title>
+  <para>
+   To work with <command>etcdctl</command>, you have to pass parameters
+   with paths to required certificates.
+  </para>
+  <para>
+   Use SSH to log into one of the cluster nodes, for example your first master.
+   The following command provides an example for using <command>etcdctl</command>.
+  </para>
+<screen>&prompt.root;<command>set -a</command>
+&prompt.root;<command>source /etc/sysconfig/etcdctl</command>
+&prompt.root;<command>etcdctl --endpoints $ETCDCTL_ENDPOINT --ca-file $ETCDCTL_CA_FILE \
+    --cert-file $ETCDCTL_CERT_FILE --key-file $ETCDCTL_KEY_FILE cluster-health</command>
+&prompt.root;<command>etcdctl cluster-health</command></screen>
+ </sect1>
+
+ <sect1 xml:id="sec.admin.troubleshooting.velum_ptf">
+  <title>Locking Installed Program Temporary Fixes</title>
+  <para>
+   It can happen that <command>zypper</command> removes an installed
+   <emphasis>Program Temporary Fix</emphasis> (<emphasis>PTF</emphasis>)
+   when updates are started with Velum. To prevent this, execute the
+   following procedure.
+  </para>
+  <procedure>
+   <step>
+    <para>
+     Install the PTF manually.
+    </para>
+    <screen>&prompt.root;<command>transactional-update reboot <replaceable>PTF_FILE</replaceable>.rpm install
+</command></screen>
+   </step>
+   <step>
+    <para>
+     As soon as the node has restarted, verify that the RPM is installed.
+    </para>
+    <screen>&prompt.root;<command>rpm -qa | grep PTF</command></screen>
+   </step>
+   <step>
+    <para>
+     Create a <command>zypper</command> lock for this RPM.
```

Comment:
I think it's very important to note here that once you `zypper al`, you'll get no updates at all on this package on the future. The user will need to eventually `zypper rl` if they want that package to be upgraded again in the future.